### PR TITLE
Closest time

### PR DESCRIPTION
**Pull request by yxdunc**

### CODE REVIEW

File: snipsowm/ontology/weather_condition.py [hunk 1]
```diff
@@ -28,12 +37,12 @@ def __init__(self, key):
...
         WeatherConditions.SNOW : ["snow is expected","it's going to snow","expect snow"],
         WeatherConditions.FOG : ["fog is expected","it's going to be foggy","expect fog"],
-        WeatherConditions.SUN : ["sun is expected","it's going to be sunny"],
+        WeatherConditions.SUN : ["sun is expected","it's going to be sunny", "the sun will shine"],
         WeatherConditions.CLOUDS : ["it's going to be cloudy","expect clouds"],
         WeatherConditions.STORM : ["storms are expected","it's going to be stormy","expect storms"],
         WeatherConditions.HUMID : ["humidity is expected","it's going to be humid"],
         WeatherConditions.WIND : ["wind is expected","it's going to be windy","expect wind"],
-        WeatherConditions.UNKNOWN : [""]
+        WeatherConditions.UNKNOWN : ["The weather is undescribable"]
```
[the remark this answers is not in the document]
Comment (michaelfester) [reply snipsowm/ontology/weather_condition.py]: WeatherCondition.UNKNOWN should probably not have a phrase attached to it. We should rather catch this as an error somewhere else, and provide a proper, streamlined error response when we don't have an appropriate weather response to speak.

File: Snipsspec [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        time = intent.weatherForecastStartDatetime
+        condition = intent.weatherForecastConditionName
+
+        skill.execute(location, time, condition)
+      %}
+  - intent: SearchWeatherForecastItem
+    action: >
+      {%
+      skill.light_on_set("white", intent.number, intent.house_room_EN)
```
[the remark this answers is not in the document]
Comment (yxdunc) [reply Snipsspec]: Yes tout le snipspec est encore a faire. J'ai besoin d'etre sur master pour le tester :)

File: snipsowm/sentence_generation/sentence_generator.py [hunk 3]
```diff
@@ -73,10 +73,62 @@ def generate_condition_sentence(tone, condition_description, locality, date):
...
+    random.shuffle(time_place_parameters)
+
+    time_place_str = time_place_parameters[0] + " " +time_place_parameters[1]
+
+    temp_time_place_params = [time_place_str, temperature]
+    random.shuffle(temp_time_place_params)
+
+    return "{} {} {}".format(sentence_beginning, *temp_time_place_params)
+
+def _flatten(array):
```
[the remark this answers is not in the document]
Comment (yxdunc) [reply snipsowm/sentence_generation/sentence_generator.py]: Il faudrait un if else statement et ca commence a etre illisible... Qu'en penses tu ?
result ~= [depth1 for depth0 in array if type(depth1) == str for depth1 in depth0 ]

File: snipsowm/ontology/weather_condition.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (yxdunc) [reply snipsowm/ontology/weather_condition.py]: We mapped UNKNOWN to values that the OWM api returns but are not a straight forward/non ambiguous condition... In a next iteration we will disambiguate querys such as hot/cold/VOLCANIC_ASH. For now I think we should raise  an error here

**Pull request merged**